ml-agents/mlagents/trainers/saver/saver.py

```diff
@@ -0,0 +1,39 @@
+# # Unity ML-Agents Toolkit
+import abc
+from mlagents.trainers.policy import Policy
+
+
+class BaseSaver(abc.ABC):
+    """This class is the base class for the Saver"""
+
+    def __init__(self):
+        pass
+
+    @abc.abstractmethod
+    def register(self, module: object, init_or_load: bool) -> None:
```

Comment:
```suggestion
    def register(self, module: Any, init_or_load: bool) -> None:
```
use `typing.Any` for unknown types. But if there are multiple types that could get registered, I think one method for each, like `register_policy()`, `register_optimizer()` is clearer.

Reply:
I want to remain the flexibility to accept things other than policy and optimizer in case in future we have some new things needed to be saved. I'll change it to `Any`.